src/utils.h

```diff
@@ -53,4 +53,19 @@ class TemporaryDirectory : boost::noncopyable {
   boost::filesystem::path tmp_name_;
 };
 
+// helper template for C (mostly openssl) data structured
+//   user should still take care about the order of destruction
+//   by instantiating StructGuard<> in a right order.
+//   BTW local variables are destructed in reverse order of instantiation
+template <typename T, void (*destroy)(T *)>
```

Comment:
`unique_ptr` can serve the same purpose and is also non-copyable (see the second optional template argument http://en.cppreference.com/w/cpp/memory/unique_ptr).

`StructGuard` could be a simple wrapper to `unique_ptr` as well.

Reply:
@lbonn Cool, thank you.